File: frontend/src/types/archive.ts
import { PaginatedResponse } from './common';

export interface ArchiveListItem {
  id: string;
  scan_id: string;
  branch?: string;
  commit_hash?: string;
  scan_created_at?: string;
  archived_at: string;
  compressed_size_bytes: number;
  findings_count: number;
  critical_findings_count: number;
  high_findings_count: number;
  dependencies_count: number;
  sbom_filenames: string[];
}

export interface AdminArchiveListItem extends ArchiveListItem {
  project_id: string;
  project_name?: string;
}

export type ArchiveListResponse = PaginatedResponse<ArchiveListItem>;
export type AdminArchiveListResponse = PaginatedResponse<AdminArchiveListItem>;

export interface ArchiveRestoreResponse {
  scan_id: string;
  project_id: string;
  message: string;
  collections_restored: string[];
}

export interface ArchiveFilters {
  branch?: string;
  date_from?: string;
  date_to?: string;
}

File: backend/app/schemas/archive.py
from datetime import datetime

from pydantic import BaseModel, Field


class ArchiveListItem(BaseModel):
    """Single archive entry in listing response."""

    id: str
    scan_id: str
    branch: str | None = None
    commit_hash: str | None = None
    scan_created_at: datetime | None = None
    archived_at: datetime
    compressed_size_bytes: int | None = None
    findings_count: int = 0
    critical_findings_count: int = 0
    high_findings_count: int = 0
    dependencies_count: int = 0
    sbom_filenames: list[str] = Field(default_factory=list)


class AdminArchiveListItem(ArchiveListItem):
    """Archive entry with project info for admin overview."""

    project_id: str
    project_name: str | None = None


class ArchiveListResponse(BaseModel):
    """Paginated list of archives for a project."""

    items: list[ArchiveListItem]
    total: int
    page: int
    size: int
    pages: int


class AdminArchiveListResponse(BaseModel):
    """Paginated list of archives across all projects (admin)."""

    items: list[AdminArchiveListItem]
    total: int
    page: int
    size: int
    pages: int


class ArchiveRestoreResponse(BaseModel):
    """Response after restoring an archive."""

    scan_id: str
    project_id: str
    message: str = "Archive restored successfully"
    collections_restored: list[str] = Field(default_factory=list)


class ScanPinResponse(BaseModel):
    """Response after pinning/unpinning a scan."""

    scan_id: str
    pinned: bool
